tty.c

```diff
@@ -154,8 +153,8 @@ bool tty_parent_select(pid_t pid, int *pwaitflags) {
 		}
 		err(1, "pselect");
 	}
-	struct timespec immediate = {0};
-	if (pselect(info.nfds, NULL, &writeFds, NULL, &immediate, &noSignals) <= 0) {
+	struct timeval immediate = {0};
+	if (select(info.nfds, NULL, &writeFds, NULL, &immediate) < 0) {
 		return false;
 	}
 	if (FD_ISSET(STDIN_FILENO, &readFds) && FD_ISSET((unsigned long) info.termfd, &writeFds)) {
```

Comment:
Is it possible for none of the write fds to be ready and we end up in a busy loop? From my tests it seems like termfd is pretty much always ready, but maybe I haven't tried hard enough to fill the pty buffer until it blocks

Reply:
It is possible.  Looks like the buffer size is ~19k in my bus.  I'm not sure what the best behavior would be in that case.

Reply:
This is very hand-wavy, but we could perform only one (blocking) select, and right after it remove all file descriptors that are ready from the watched sets, before re-adding them when we get through the conditions:

```c
int rc = select(info.nfds, &readFds, &writeFds, NULL, NULL) == -1);
...
// conceptually; doesn't actually work because fd_set isn't a number
info.rfds &= ~readFds;
info.wfds &= ~writeFds;
...
if (FD_ISSET(STDIN_FILENO, &readFds) && FD_ISSET((unsigned long) info.termfd, &writeFds)) {
    ...
    FD_SET(STDIN_FILENO, &info.rfds);
    FD_SET((unsigned long) info.termfd, &info.wfds);
}
```

Given the fd_set shenanigans to remove the complement of the set, it might be easier to use poll() instead.

If this works, we can do this here or in another commit, your call.

Reply:
I think I'll tackle this in a follow on commit.  I already have the poll changes since it was something I tried when debugging the signal issue.  I think what we want to do is remove stdin and add term(w) to the poll list if the term is not writeable.  Then we can flip it back when the term becomes writeable again.  We could do the same with term(r) and stdout.